src/transport/abstract-transport.js

```diff
@@ -20,6 +20,7 @@ function AbstractTransport(address, timeout, messageHandler) {
   this.awaitingPump = false;
   this.connected = false;
   this.backoffMilliseconds = 1;
+  this.nextReconnectListenerToken = 1;
```

Comment:
Can you explain how multiple reconnect tokens are used here?  It seems (naively?) possible to have just a single block subscription property, which is set to the subscription ID, and is replaced any time reconnect happens.  When would it be necessary to track multiple newHeads subscriptions?